readmes/custom_viz.md

```diff
@@ -0,0 +1,289 @@
+
+### Drawing the UI Window
+
+In the Visualization Package, UI windows are registered based on its topic, and will update whenever a message is sent or received on it. That being said, the visualizer base class (in this case, `HistoryDrawingVisualizer`) manages all of this already. All we'll have to do is format the message contents in the window.
+
+- Implementations of the visualizer classes override the function `CreateGUI()` in order to populate the UI window. By default, this function will simply display the raw message converted to a string. The HistoryDrawingVisualizer acts a bit differently, managing multiple messages over time.
+
+    Begin by adding an empty override for HistoryDrawingVisualizer's `CreateGUI` function to your visualizer script.
+
+    ```csharp
+    public override Action CreateGUI(IEnumerable<Tuple<PoseStampedMsg, MessageMetadata>> messages)
+    {
+        return () =>
+        {
+
+        };
+    }
+    ```
+
+- The Visualization Package contains a convenient set of utility functions to format common message types to strings as well as draw common geometries. As the messages are passed in as an IEnumerable, we can simply iterate through them and call the desired functions.
+
+    As `/goal_pose` is a PoseStampedMsg, we will use the utility function for pose GUIs, as defined [here](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/bc6c4220d39ced0e17c64553917357d6c90fb3f2/com.unity.robotics.visualizations/Runtime/Scripts/VisualizationUtils.cs#L338). The function takes an optional string parameter that serves as a label for the formatting. In this case, label the string as with a goal number, counting up from a newly initialized counter variable.
+
+    Add this functionality between the `return` curly brackets. Your overridden function should look as follows:
+
+    ```csharp
+    public override Action CreateGUI(IEnumerable<Tuple<PoseStampedMsg, MessageMetadata>> messages)
+    {
+        return () =>
+        {
+            int count = 0;
+            foreach (var (message, meta) in messages)
+            {
+                message.pose.GUI($"Goal #{count}");
+                count++;
+            }
+        };
+    }
+    ```
+
+- The GUI window is almost ready to be tested! Open a terminal window in your ROS workspace. Source the workspace and, once again, run the `ros2 launch` for the project if it is not already running.
+
+    ```bash
+    ros2 launch unity_slam_example unity_slam_example.py
+    ```
+
+- Return to Unity. Select the `TrailVisualizer` object, and in the Inspector, find this `PoseTrailVisualizer` component. 
+
+    Change the `History Length` field to however many goals you would like to track over time--for example, `5`.
+
+- Before we test it, we'll want to set this as the new default visualizer for `/goal_pose` messages. In your `TrailVisualizer` object, add the `Priority Setter` component. 
```

Comment:
Do we not need to set the topic to /goal_pose in the component?

Reply:
you don't have to! because it's specified to be <PoseStampedMsg>, it will have this behavior for all pose stamped messages, including goal pose--i can clarify that as well